src/Discord.Net.SlashCommands/SlashCommandService.cs

```diff
@@ -0,0 +1,675 @@
+        internal readonly string _wildCardExp;
+        internal readonly RunMode _runMode;
+
+        /// <summary>
+        /// Represents all of the modules that are loaded in the <see cref="SlashCommandService"/>
+        /// </summary>
+        public IReadOnlyList<ModuleInfo> Modules => _moduleDefs.ToList();
+
+        /// <summary>
+        /// Get all of the executeable Slash Commands that are loaded in the <see cref="SlashCommandService"/> modules
+        /// </summary>
+        public IReadOnlyList<SlashCommandInfo> SlashCommands => _moduleDefs.SelectMany(x => x.SlashCommands).ToList();
+
+        /// <summary>
+        /// Get all of the executeable Context Commands that are loaded in the <see cref="SlashCommandService"/> modules
+        /// </summary>
+        public IReadOnlyList<ContextCommandInfo> ContextCommands => _moduleDefs.SelectMany(x => x.ContextCommands).ToList();
+
+        /// <summary>
+        /// Get all of the Interaction handlers that are loaded in the <see cref="SlashCommandService"/>
+        /// </summary>
+        public IReadOnlyCollection<InteractionInfo> Interacions => _moduleDefs.SelectMany(x => x.Interactions).ToList();
+
+        /// <summary>
+        /// Client that the Application Commands will be registered for
+        /// </summary>
+        public BaseSocketClient Client { get; }
+
+        /// <summary>
+        /// Initialize a <see cref="SlashCommandService"/> with the default configurations
+        /// </summary>
+        /// <param name="discord">The client that will be used to register commands</param>
+        public SlashCommandService (BaseSocketClient discord) : this(discord, new SlashCommandServiceConfig()) { }
+
+        /// <summary>
+        /// Initialize a <see cref="SlashCommandService"/> with configurations from a provided <see cref="SlashCommandServiceConfig"/>
+        /// </summary>
+        /// <param name="discord">The client that will be used to register commands</param>
+        /// <param name="config">The configuration class</param>
+        public SlashCommandService (BaseSocketClient discord, SlashCommandServiceConfig config)
+        {
+            _lock = new SemaphoreSlim(1, 1);
+            _typedModuleDefs = new ConcurrentDictionary<Type, ModuleInfo>();
+            _moduleDefs = new HashSet<ModuleInfo>();
+
+            _logManager = new LogManager(config.LogLevel);
+            _logManager.Message += async msg => await _logEvent.InvokeAsync(msg).ConfigureAwait(false);
+            _cmdLogger = _logManager.CreateLogger("App Commands");
+
+            _slashCommandMap = new SlashCommandMap<SlashCommandInfo>(this);
+            _contextCommandMap = new SlashCommandMap<ContextCommandInfo>(this);
+            _interactionCommandMap = new SlashCommandMap<InteractionInfo>(this, config.InteractionCustomIdDelimiters);
+
+            Client = discord;
+
+            _runMode = config.RunMode;
+            if (_runMode == RunMode.Default)
```

Comment:
redundant

Reply:
It is needed to change the command run mode from the command attribute.
[Commands](https://github.com/Discord-Net-Labs/Discord.Net-Labs/blob/release/3.x/src/Discord.Net.Commands/CommandService.cs#L101-L102)